### ISSUE TITLE
Support client certificates

### ISSUE THREAD

**Issue by lkinley-rythmos** [ballerina-platform/module-ballerinax-mongodb#81]
**Description:**

There appears to be no support for using client certificates for connecting to the database.  In considering using ballerina this is a deal-breaker for us.  If it is indeed supported, please document how it can be done.
**Suggested Labels:**

enhancement
**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

**Comment by niveathika** [comment]
@lkinley-rythmos The connector does support connecting with SSL certificates.  You can enable this by using ConnectionProperties.sslEnabled (https://github.com/ballerina-platform/module-ballerinax-mongodb/blob/master/mongodb/src/mongodb/client.bal#L109)

**Comment by lkinley-rythmos** [comment]
How do I specify the path the client certificate, CA certificate, key, key password, etc?  I don't see provision for this in the code.

**Comment by niveathika** [comment]
@lkinley-rythmos Seems you are right. The certificates are taken from Default SSLContext in Java runtime.

@daneshk, @Bhashinee  How about we introduce these configs in the connection properties as follows, 
```ballerina
public type ConnectionProperties record {|
    ....
    KeyStore? keyStore = ();
    TrustStore? trustStore = ();
|}

public type KeyStore record {|
    string path;
    string password;
|};

public type TrustStore record {|
    string path;
    string password;
|};

```
With these configs we can pass custom SSL context to MongoDB connection.

**Comment by daneshk** [comment]
@niveathika Shall we support it like,

```ballerina
public type ConnectionProperties record {|
    ....
    SecureSocket? secureSocket = ();
|}

# Configurations related to facilitating secure communication with a remote endpoint.
#
# + trustStore - Configurations associated with the TrustStore
# + keyStore - Configurations associated with the KeyStore
# + protocol - The standard name of the requested protocol
public type SecureSocket record {|
    crypto:TrustStore? trustStore = ();
    crypto:KeyStore? keyStore = ();
    string protocol = "TLS";
|};
```

We already have config record in the `crypto` module, We can reuse it.

**Comment by niveathika** [comment]
@daneshk Ack

**Comment by niveathika** [comment]
@lkinley-rythmos We have released patch version v1.0.1(based on ballerina v1.2.x) & v2.0.1(based on ballerina Swan lake preview 1) with this fix.

Thank you for reporting this issue. Please feel free to reach out if you face any issues.
